plenoptic/simulate/models/frontend.py

```diff
@@ -1,96 +1,407 @@
+        return self.surround_std.clamp(
+            min=self.width_ratio_limit * float(self.center_std), max=None
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = same_padding(x, self.kernel_size, pad_mode=self.pad_mode)
+        self._clamp_surround_std()  # clip the surround stdev
+        filt = self.get_filter()
+        y = F.conv2d(x, filt, bias=None)
+        return y
+
+
+class LN(nn.Module):
+    """Linear-Nonlinear model, applies a difference of Gaussians filter followed by an
+    activation function.
+
+    Parameters
+    ----------
+    activation:
+        Activation function following linear convolution.
+
+    Attributes
+    ----------
+    center_surround: nn.Module
+        `CenterSurround` difference of Gaussians filter.
 
     Notes
     -----
-    Berardino et al., Eigen-Distortions of Hierarchical Representations (2017)
-    http://www.cns.nyu.edu/~lcv/eigendistortions/ModelsIQA.html
+    See `CenterSurround` class for full Parameter docstring.
+
     """
 
-    def __init__(self, disk_mask=False, pretrained=False, requires_grad=True):
-        """
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, int]],
+        center: str = "on",
+        width_ratio_limit: float = 4.0,
+        amplitude_ratio: float = 1.25,
+        pad_mode: str = "circular",
+        activation: Callable[[Tensor], Tensor] = F.softplus,
+    ):
+        super().__init__()
+        self.center_surround = CenterSurround(
+            kernel_size,
+            center,
+            width_ratio_limit,
+            amplitude_ratio,
+            pad_mode=pad_mode,
+        )
+        self.activation = activation
 
-        """
+    def forward(self, x: Tensor) -> Tensor:
+        y = self.activation(self.center_surround(x))
+        return y
+
+
+class LG(nn.Module):
+    """ Linear center-surround followed by luminance gain control and activation.
+    Parameters
+    ----------
+    See `CenterSurround` class for full Parameter docstring.
+
+    Attributes
+    ----------
+    center_surround: nn.Module
+        Difference of Gaussians linear filter.
+    luminance: nn.Module
+        Gaussian convolutional kernel used to normalize signal by local luminance.
+    luminance_scalar: nn.Parameter
+        Scale factor for luminance normalization.
+    """
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, int]],
+        center: str = "on",
+        width_ratio_limit: float = 4.0,
+        amplitude_ratio: float = 1.25,
+        pad_mode: str = "circular",
+        activation: Callable[[Tensor], Tensor] = F.softplus,
+    ):
         super().__init__()
+        self.center_surround = CenterSurround(
+            kernel_size,
+            center,
+            width_ratio_limit,
+            amplitude_ratio,
+            pad_mode=pad_mode,
+        )
+        self.luminance = Gaussian(kernel_size=kernel_size)
+        self.luminance_scalar = nn.Parameter(torch.rand(1) * 10)
```

Comment:
I'm a bit confused by this. Does this initialization lead to good behavior? I thought this number was supposed to be relatively small

Reply:
The "true" learned values are 5ish and 15ish. These can be seen in the OnOff default statedict. 

These scalars multiply the filter responses and are not additive constants (e.g. semi-saturation constants), in case that might be what you're thinking of.

Reply:
Yup, that was exactly what I was thinking of.